internal/util/util.go

```diff
@@ -433,3 +445,18 @@ func recreateVolumeSnapshotContent(vsc snapshotv1api.VolumeSnapshotContent, back
 
 	return nil
 }
+
+func DeleteVolumeSnapshotIfAny(ctx context.Context, snapshotClient snapshotterClientSet.Interface,
+	vs snapshotv1api.VolumeSnapshot, log logrus.FieldLogger) {
+	_, err := snapshotClient.SnapshotV1().VolumeSnapshots(vs.Namespace).Get(ctx, vs.Name, metav1.GetOptions{})
```

Comment:
This can be simplified --- directly call `snapshotClient.SnapshotV1().VolumeSnapshots(vs.Namespace).Delete` and check for `apierrors.IsNotFound(err)` and log messages accordingly.